### PR TITLE
Add MountedKey encryption.Key implementation

### LINKED ISSUE #19134: Create 'mounted key' encryption.Key implementation
The most general purpose encryption.Key we can make is one that mounts a key from the environment, and encrypts data locally, rather than using the APIs provided by GCloud, AWS, or Vault.

There are a few ways we could load the key:
* Mounted file path: This is probably the most secure way, and is compatible with the way kubernetes provides managed secrets
* Environment variable: This feels less secure, it trades off a little deployment security for convenience, some users might want this
* Direct key in config: i don't think we should do this, as it'll be exposed in the admin UI

### PR DESCRIPTION
**Pull request by arussellsaw**
This PR adds a new 'mounted' encryption.Key implementation that loads an encryption key either from an environment variable or a file, this is intended for users that either don't have access to or don't want to use cloud encryption APIs

this was co-authored with @indradhanush a week or two ago, just getting round to adding tests and PRing now.

closes #19134

### CODE REVIEW

File: internal/encryption/mounted/key.go [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+	"os"
+	"strings"
+
+	"github.com/cockroachdb/errors"
+
+	"github.com/sourcegraph/sourcegraph/internal/encryption"
+	"github.com/sourcegraph/sourcegraph/schema"
+)
+
+type Key struct {
```
Comment (unknwon) [inline internal/encryption/mounted/key.go:23]: Prefer adding some docstring here :)

File: internal/encryption/mounted/key_test.go [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+import (
+	"context"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"k8s.io/apimachinery/pkg/util/rand"
+
+	"github.com/stretchr/testify/require"
```
Comment (unknwon) [inline internal/encryption/mounted/key_test.go:12]:
```suggestion
	"github.com/stretchr/testify/assert"
	"k8s.io/apimachinery/pkg/util/rand"
	"github.com/stretchr/testify/require"
```

(may need re-sort)

**Review by unknwon — APPROVED** [review]
Approve to unblock

File: internal/encryption/mounted/key.go [hunk 3]
```diff
@@ -0,0 +1,143 @@
+package mounted
+
+import "C"
```
Comment (asdine) [inline internal/encryption/mounted/key.go:3]: I'm curious about this, does one of the encryption package need it?

**Review by asdine — APPROVED** [review]
I think a few docstrings are missing, otherwise LGTM

File: internal/encryption/mounted/key.go (hunk 3 above)
Comment (arussellsaw) [reply internal/encryption/mounted/key.go:3]: oh weird, not sure how that snuck in. tests pass without it so maybe not?

**Pull request merged**